### PR TITLE
RIOT OS support, test scripts will be submitted to RIOT-OS repository

### PR DESCRIPTION
**Pull request by kaleb-himes**
Initial testing successful. There is a lot we could do here.

The RIOT_OS make file system has a "make all" option that attempts to build for every supported platform. I have not set up that same functionality yet as there is a learning curve associated with setting that up. I have yet to conquer it. When we have more time to focus on it that would be a neat feature to implement.

List of platforms supported by RIOT OS can be found here: https://github.com/RIOT-OS/RIOT/wiki

### CODE REVIEW

**Comment by kaleb-himes** [comment]
Ready for review once tests pass. If we can merge this before end of day today I can open a PR on RIOT examples here: https://github.com/RIOT-OS/RIOT

File: wolfssl/ssl.h [hunk 1]
```diff
@@ -1065,6 +1065,8 @@ WOLFSSL_API int wolfSSL_make_eap_keys(WOLFSSL*, void* key, unsigned int len,
         #ifdef __PPU
             #include <sys/types.h>
             #include <sys/socket.h>
+        #elif defined(WOLFSSL_RIOT_OS)
+            #include <sys/_types/_iovec_t.h>
```
Comment (dgarske) [inline wolfssl/ssl.h]: This is a Linux/Mac include and doesn't appear to be portable. Have you considered using NO_WRITEV in settings.h for WOLFSSL_RIOT_OS?

File: testsuite/testsuite.c [hunk 2]
```diff
@@ -210,7 +210,11 @@ int testsuite_test(int argc, char** argv)
 #endif /* HAVE_WNR */
 
     printf("\nAll tests passed!\n");
+#ifdef WOLFSSL_RIOT_OS
+    exit(0);
+#else
     return EXIT_SUCCESS;
+#endif
 }
```
Comment (dgarske) [inline testsuite/testsuite.c]: Perhaps we can create a single macro to handle exiting tests to keep this clean?

#ifdef WOLFSSL_RIOT_OS
#define EXIT_TEST(ret) exit(ret)
#else
#define EXIT_TEST(ret) return ret;
#endif

Recommend putting this macro logic somewhere common for these like wolfssl/wolfcrypt/types.h.

File: wolfssl/ssl.h (hunk 1 above)
Comment (kaleb-himes) [inline wolfssl/ssl.h]: Great suggestion!

**Pull request merged**